File: Algorithms/p020_Valid_Parentheses/p020_Valid_Parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> my_stack;
        for(char c : s){
            if(c=='('||c=='['||c=='{'){
                my_stack.push(c);
            }else if(my_stack.empty() ||
                     c==')'&&my_stack.top()!='(' ||
                     c==']'&&my_stack.top()!='[' ||
                     c=='}'&&my_stack.top()!='{'){
                return 0;
            }else{
                my_stack.pop();
            }
        }
        return my_stack.empty();
    }
};

File: Algorithms/p020_Valid_Parentheses/p020_Valid_Parentheses.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        stack = []
        for c in s:
            if c in '([{':
                stack.append(c)
            elif not stack or \
                 c==')' and stack[-1]!='(' or \
                 c==']' and stack[-1]!='[' or \
                 c=='}' and stack[-1]!='{':
                return False
            else:
                stack.pop()
        return not stack
